source/common/router/config_impl.cc

```diff
@@ -353,7 +358,36 @@ bool RouteEntryImplBase::evaluateRuntimeMatch(const uint64_t random_value) const
                                                        random_value);
 }
 
-bool RouteEntryImplBase::matchRoute(const Http::HeaderMap& headers, uint64_t random_value) const {
+bool RouteEntryImplBase::evaluateCredentialMatch(const StreamInfo::StreamInfo& stream_info) const {
+  bool matches = true;
+
+  if (!credentialMatchCriteria()) {
+    return matches;
+  }
+
+  const CredentialMatchCriteria& criteria = *credentialMatchCriteria();
+
+  if (criteria.presented()) {
+    const bool peerPresented = stream_info.downstreamSslConnection() &&
+                               stream_info.downstreamSslConnection()->expirationPeerCertificate();
+    matches &= *criteria.presented() == peerPresented;
```

Comment:
nits: I think `has_value()` and `value()` is preferred if it is `optional<bool>`

Reply:
modified to use `has_value()` & `value()`